core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1764,17 +1764,21 @@ class SparkContext(config: SparkConf) extends Logging {
     listenerBus.post(SparkListenerUnpersistRDD(rddId))
   }
 
+  def addJar(path: String) {
+    addJar(path, false)
+  }
+
   /**
    * Adds a JAR dependency for all tasks to be executed on this `SparkContext` in the future.
    *
    * If a jar is added during execution, it will not be available until the next TaskSet starts.
    *
    * @param path can be either a local file, a file in HDFS (or other Hadoop-supported filesystems),
    * an HTTP, HTTPS or FTP URI, or local:/path for a file on every worker node.
-   *
+   * @param check when true, check jar file exists
    * @note A path can be added only once. Subsequent additions of the same path are ignored.
    */
-  def addJar(path: String) {
+  def addJar(path: String, check: Boolean) {
```

Comment:
Hm, I'm not sure about a whole new API method for this. You're just trying to make it fail faster in case of user error. What about checking if it exists and warning?

Reply:
@srowen Since if the path is wrong, all task after this will be wrong. SInce task description will bring all jar path in sparkcontext. then when executor check dependencies, it will run, Just like what I have show above. 
![image](https://user-images.githubusercontent.com/46485123/60277859-f803f680-9930-11e9-87c7-388a95d93d16.png)

For user's error, it always happen when SparkThirftServer and Spark SQL Cli, when this case, client will got error message but Core server won't fail. That's why I make it throw exception.

Reply:
I see, but can you fix that problem by making the thriftserver resilient to app errors like this?
I may be incorrect that there's any use case for adding a JAR that can't yet be accessed. If so I think you just want to unilaterally check for the existence of the JAR. This shouldn't be a config or API option.

Reply:
@srowen 
And for you concern, new method 
`def addJar(path : String ) = addJar(path, false)`
Will replace old SparkContext#addJar  method and do the same thing like before .

Reply:
Hm, Since always sql action make this error. So I just want to check for SQL AddJar behavior
If we don't check this in driver level,  executor level will return error ， if we ignore executor's error like this. User can just know it when executor can't find class.

Reply:
IMO I'm reluctant to add new API for this existence check. I would rather to do the check every time calling this `addJar` method. Since we already had this check in `addFile`, I think it is ok to keep this behavior align with the implementation of `addFile`.

Reply:
@jerryshao 
I can't promise if it can impact other process. So I do like this, ignore this in executor end is not elegant.